command/regenesis/cmd.go

```diff
@@ -0,0 +1,161 @@
+		defer outputter.WriteOutput()
+
+		if params.SnapshotTrieDBPath == "" || params.TrieDBPath == "" || params.TrieRoot == "" {
+			outputter.SetError(fmt.Errorf("not enough arguments"))
+
+			return
+		}
+
+		trieDB, err := leveldb.OpenFile(params.TrieDBPath, &opt.Options{ReadOnly: true})
+		if err != nil {
+			outputter.SetError(fmt.Errorf("open trie trieDB error:%w", err))
+
+			return
+		}
+		defer trieDB.Close()
+
+		snapshotDB, err := leveldb.OpenFile(params.SnapshotTrieDBPath, nil)
+		if err != nil {
+			outputter.SetError(fmt.Errorf("open snapshotDB error:%w", err))
+
+			return
+		}
+		defer snapshotDB.Close()
+
+		snapshotStorage := itrie.NewKV(snapshotDB)
+
+		err = itrie.CopyTrie(types.StringToHash(params.TrieRoot).Bytes(), itrie.NewKV(trieDB), snapshotStorage, nil, false)
+		if err != nil {
+			outputter.SetError(fmt.Errorf("copy trie error:%w", err))
+
+			return
+		}
+
+		checkedHash, err := itrie.HashChecker(types.StringToHash(params.TrieRoot).Bytes(), snapshotStorage)
+		if err != nil {
+			outputter.SetError(fmt.Errorf("copy trie error:%w", err))
+
+			return
+		}
+
+		if checkedHash != types.StringToHash(params.TrieRoot) {
+			outputter.SetError(fmt.Errorf("incorrect trie root error:%w", err))
+
+			return
+		}
+
+		outputter.WriteCommandResult(&ReGenesisResult{})
+	}
+
+	genesisCmd.Flags().StringVar(
+		&params.SnapshotTrieDBPath,
+		"snapshotPath",
+		"",
+		"the directory of trie data of trie copy",
+	)
+	genesisCmd.Flags().StringVar(
+		&params.TrieDBPath,
+		"triedb",
+		"",
+		"the directory of trie data of old chain",
+	)
+	genesisCmd.Flags().StringVar(
+		&params.TrieRoot,
+		"stateRoot",
+		"",
+		"block state root of old chain",
+	)
+
+	getRootCmd := &cobra.Command{
+		Use:   "getroot",
+		Short: "returns state root of old chain",
+		Run: func(cmd *cobra.Command, args []string) {
+			outputter := command.InitializeOutputter(genesisCmd)
+			defer outputter.WriteOutput()
+
+			rpcClient, err := jsonrpc.NewClient(params.JSONRPCAddress)
+			if err != nil {
+				outputter.SetError(fmt.Errorf("connect to client error:%w", err))
+
+				return
+			}
+
+			block, err := rpcClient.Eth().GetBlockByNumber(ethgo.BlockNumber(params.BlockNumber), false)
+			if err != nil {
+				outputter.SetError(fmt.Errorf("get block error:%w", err))
+
+				return
+			}
+
+			outputter.WriteCommandResult(&ReGenesisResult{
+				Message: fmt.Sprintf("state root %s for block %d", block.StateRoot, block.Number),
+			})
+		},
+	}
+
+	genesisCmd.AddCommand(getRootCmd)
+	getRootCmd.Flags().StringVar(
+		&params.JSONRPCAddress,
+		"rpc",
+		"",
+		"the JSON RPC IP address for old chain",
+	)
+	getRootCmd.Flags().Int64Var(
+		&params.BlockNumber,
+		"block",
+		int64(ethgo.Latest),
+		"Block number of trie snapshot",
+	)
+
+	return genesisCmd
+}
+
+type ReGenesisResult struct {
+	Message string `json:"message"`
+}
+
+func (r *ReGenesisResult) GetOutput() string {
+	var buffer bytes.Buffer
+
+	buffer.WriteString("\n[Trie copy SUCCESS]\n")
```

Comment:
Would be great to give more information in the output, like the target path, and some details on the performed actions.